### DISCUSSION TITLE
Selecting a row on Many to Many field

### DISCUSSION THREAD

**Discussion by lambdina** [supabase/supabase#15079] [Questions]
Hi everyone.

I want to query a row with a **specific** field. Not that difficult, I should use `eq`, right.
The problem is when doing nesting queries with **Many to Many** relationship fields. Let me illustrate :

My tables are configured the following way:
```
create table products (
  "id" serial primary key,
  "name" text,
  "description" text,
  "price" float,
);

create table p_variable (
  "id" serial primary key,
  "color" text,
  "size" text,
  "gender" text,
  "nb_stocks" text
);

create table category (
  "id" serial primary key,
  "name" text,
  "parent" int references category
);

create table categories (
  "p_variable_id" int references p_variable,
  "category_id" int references category,
  primary key (p_variable_id, category_id)
);
```

As you may have noticed I followed the example in order to create many to many relationship between `category` and product, because a product can be on several categories, and a category contains several products.

My question is, when I want to query products from a specific categoryId, how should I do ?

I tried the following:
```javascript
                const { data, error } = await supabase
                                .from('p_variable')
                                .select(`id,images,nb_stock,product ( id,name, price, category (id,name, parent) )`)
                                .eq(`id, ${categoryId}`)
                                .range(to, from);
```
(with to and range being there for pagination purposes.

But obviously it's not going to work since there are several `id`s in my query.
Any ideas ?
Thanks in advance !

**Comment by GaryAustin1** [comment]
```
 .eq(`id, ${categoryId}`)
```
Is not correct formatting for supabase-js.
`.eq('id', categoryId)` is with two separate parameters.

Also it is likely either 'product.category.id' or 'category.id' to get the id you want.
